source/material_model/steinberger.cc

```diff
@@ -495,50 +487,49 @@ namespace aspect
           conductivity_transition_depths = Utilities::string_to_double
                                            (Utilities::split_string_list(prm.get ("Thermal conductivity transition depths")));
           const unsigned int n_conductivity_layers = conductivity_transition_depths.size() + 1;
+
           AssertThrow (std::is_sorted(conductivity_transition_depths.begin(), conductivity_transition_depths.end()),
                        ExcMessage("The list of 'Thermal conductivity transition depths' must "
                                   "be sorted such that the values increase monotonically."));
 
-          reference_thermal_conductivities = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Reference thermal conductivities"))),
-                                                                                     n_conductivity_layers,
-                                                                                     "Reference thermal conductivities");
-          conductivity_pressure_dependencies = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Pressure dependencies of thermal conductivity"))),
-                                                                                       n_conductivity_layers,
-                                                                                       "Pressure dependencies of thermal conductivity");
-          conductivity_reference_temperatures = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Reference temperatures for thermal conductivity"))),
-                                                                                        n_conductivity_layers,
-                                                                                        "Reference temperatures for thermal conductivity");
-          conductivity_exponents = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Thermal conductivity exponents"))),
-                                                                           n_conductivity_layers,
-                                                                           "Thermal conductivity exponents");
-          saturation_scaling = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Saturation prefactors"))),
-                                                                       n_conductivity_layers,
-                                                                       "Saturation prefactors");
```

Comment:
I do not quite see the benefit of changing this from a list to parsing a map. Is there any use case we do not support with the current version or something we can do cleaner with the map parsing? I dont think we have to convert all input parameter parsing to the map, unless there is a benefit to it.

Reply:
The only benefit at the moment is that (I think) it looks a bit cleaner. Maybe in the future someone will want to copy this into another model that might need multiple phases?

Reply:
I've restored the original format, which reduces the line changes.